api/services/protocols/ldap.js

```diff
@@ -0,0 +1,73 @@
+var _ = require('lodash');
+var adminGroup = new RegExp(process.env.ADMIN_GROUP_REG || null);
+var commonName = /^cn=([^,]+),.*/
+
+var ldapToUser = function (ldapUser, next) {
+    var data = {
+        active: true
+    }
+    data.username = ldapUser.uid;
+    data.firstName = ldapUser.givenName;
+    data.lastName = ldapUser.sn;
+    data.email = ldapUser.mail;
+
+    sails.models.user.create(data)
+        .exec(function (err, user) {
+            if (err) {
+                next(err);
+            } else {
+                adminStatus(ldapUser, user, next);
+            }
+        });
+}
+
+var group_test = function (group) {
+    return group.cn === 'admin' || adminGroup.test(group.cn);
+}
+
+var member_test = function (group) {
+     return group.startsWith('cn=admin') ||
+         adminGroup.test(commonName.replace(group, "$1"));
+}
+
+var adminStatus = function (ldapUser, user, next) {
+    user.admin =
+        _.findIndex(ldapUser._groups, group_test) > -1 ||
+        _.findIndex(ldapUser.memberOf, member_test) > -1;
+    next(null, user);
+}
+
+/**
+ * Resolve LDAP user
+ *
+ * This function can be used to create a user in the local db to store
+ * users' roles locally
+ *
+ * @param {Request}   request
+ * @param {Response}  response
+ * @param {Function}  next
+ */
+exports.resolve = function resolve(request, response, next) {
```

Comment:
Since these aren't really used, can you instead assign the resolveUser function?

Reply:
`next` is used. I could probably get rid of request and response, but that pattern appears over and over again in this code even when they're not being used. `resolveUser` is the "next" to the authenticate, we then need to call the original next to finalize the login and session creation.